rustworkx-core/src/token_swapper.rs

```diff
@@ -205,6 +205,16 @@ where
         }
         let id_node = self.rev_node_map[&node];
         let id_token = self.rev_node_map[&tokens[&node]];
+
+        if self
+            .graph
+            .neighbors(id_node)
+            .collect::<Vec<G::NodeId>>()
+            .is_empty()
```

Comment:
```suggestion
            .neighbors(id_node)
            .next()
            .is_some()
```

Reply:
Oh, this is how one can directly check whether an iterator is empty or non-empty, I was wondering about that. Fixed in [d682b9b](https://github.com/Qiskit/rustworkx/pull/995/commits/d682b9ba5198e7a1d4c371fb494c41123cbab18b).